### DISCUSSION TITLE
Can't run : illegal hardware instruction

### DISCUSSION THREAD

**Discussion by Andrufus** [ghostty-org/ghostty#4210] [Q&A] [answered]
Hi everyone !

So, I have this old laptop from 2011. Just wanted to try ghostty. Well it was a rather short experience because it just doesn't run :smile: 
Here are the specs:
- Distro: Fedora Linux 40 (Cinnamon)
- Kernel: 6.12.5-100.fc40.x86_64
- CPU: Intel Core i3-2310M
- Graphics:
  - Device-1: Intel HD Graphics 3000
  - Device-2: NVIDIA GF119M [GeForce GT 520MX]
  - Display: x11

I installed it via Terra as per the [docs](https://ghostty.org/docs/install/binary#fedora) .
I tried to launch it from the Cinnamon menu and nothing happened.
Then, I tried from a terminal and I got the error from the title : `illegal hardware instruction`.

I thought maybe it's a GPU problem(since there are 2 in my laptop) so I tried to run it forcing the use of the nvidia GPU with this command : `witcherooctl launch --gpu=1 ghostty` (nvidia gpu is listed device 1 with switcherooctl list). Same error.

So, is my laptop too old for this ?

**Comment by rhodes-b** [comment]
ghostty tries to do some SIMD stufff I wonder if that is what is happening here (It should only try them if it detects your cpu to have them though so that logic might be incorrect)

**Comment by jcollie** [comment]
How did you compile Ghostty? Zig can be aggressive about using CPU features of the system it's compiled on. If you then move the binary to another system you get illegal instructions because it's trying to use CPU features that aren't present. For binaries that are going to be shared across systems we recommend using `-Dcpu=baseline` to tell Zig not to use non-portable instructions.

The SIMD parts of Ghostty use runtime CPU detection to load the appropriate binary code for that CPU architecture so it's unlikely to be the cause of illegal instructions.

**Reply by rhodes-b** [reply]
https://github.com/terrapkg/packages/blob/frawhide/anda/devs/ghostty/stable/ghostty.spec#L84-L89
they said they got it from Terra, which doesnt have the baseline flag

**Comment by rhodes-b** [comment, marked as answer]
I added the patch in a pr for you, hopefully that should fix it
https://github.com/terrapkg/packages/pull/2802

**Reply by Andrufus** [reply]
Woah, that was fast !
It does indeed fix the problem but it seems now I have [this problem](https://github.com/ghostty-org/ghostty/discussions/3882).
I guess I will just have to wait for the 1.0.1 and see if it works

**Reply by Andrufus** [reply]
It's working fine now with version 1.0.1. Thanks !
